cli/cmd/root.go

```diff
@@ -150,21 +173,66 @@ func init() {
+	rootCmd.PersistentFlags().StringVar(&scanConfigName, "scan-config-name", "", "use an existing scan config that is defined in the VMClarity server")
 	rootCmd.PersistentFlags().StringVar(&scanResultID, "scan-result-id", "", "the ScanResult ID to export the scan results to")
 	rootCmd.PersistentFlags().BoolVar(&mountVolume, "mount-attached-volume", false, "discover for an attached volume and mount it before the scan")
 	rootCmd.PersistentFlags().BoolVar(&waitForServerAttached, "wait-for-server-attached", false, "wait for the VMClarity server to attach the volume")
+	rootCmd.PersistentFlags().StringVar(&input, "input", "", "input for families")
+	rootCmd.PersistentFlags().StringVar(&inputType, "input-type", "dir", "input type for families")
+	rootCmd.PersistentFlags().BoolVar(&cicdMode, "cicd-mode", false, "CICD mode")
+	rootCmd.PersistentFlags().BoolVar(&exportCICDResults, "export-cicd-results", false, "export results to VMclarity server")
 
 	// TODO(sambetts) we may have to change this to our own validation when
 	// we add the CI/CD scenario and there isn't an existing scan-result-id
 	// in the backend to PATCH
-	rootCmd.MarkFlagsRequiredTogether("server", "scan-result-id")
+	validateRequiredFlagForDefinedFlag(rootCmd, "scan-result-id", "server")
+	validateRequiredFlagForDefinedFlag(rootCmd, "scan-config-name", "server")
+	rootCmd.MarkFlagsMutuallyExclusive("config", "scan-config-name")
+	rootCmd.MarkFlagsMutuallyExclusive("mount-attached-volume", "input")
+	validateRequiredFlagForDefinedFlag(rootCmd, "input-type", "input")
+	validateRequiredFlagForDefinedFlag(rootCmd, "export-cicd-results", "server")
 }
 
-// initConfig reads in config file and ENV variables if set.
-func initConfig() {
-	logrus.Infof("init config")
+func validateRequiredFlagForDefinedFlag(rootCmd *cobra.Command, definedFlag, requiredFlag string) {
+	if flag := rootCmd.Flag(definedFlag); flag != nil {
+		if required := rootCmd.Flag(requiredFlag); required == nil {
+			panic(fmt.Sprintf("Cannot set flag '%s' alone without flag '%s'", definedFlag, requiredFlag))
+		}
+	}
+}
+
+func getConfigFromBackend() *families.Config {
+	if server == "" {
+		panic("Missing backend")
+	}
+	client, err := backendclient.Create(server)
+	if err != nil {
+		logrus.Errorf("failed to create VMClarity API client: %v", err)
+	}
+
+	scanConfigs, err := client.GetScanConfigs(context.TODO(), models.GetScanConfigsParams{
+		Filter: utils.PointerTo(fmt.Sprintf("name eq '%s'", scanConfigName)),
+	})
+	if err != nil {
+		panic(fmt.Sprintf("Failed to get scan config by name %v", err))
+	}
+	if len(*scanConfigs.Items) == 0 {
+		panic(fmt.Sprintf("There is no scan config with name=%s", scanConfigName))
+	}
+
+	scanConfig := families.CreateFamilyConfigFromModel(
+		(*scanConfigs.Items)[0].ScanFamiliesConfig,
+		families.LoadAddresses("localhost"),
+		families.LoadPaths(),
+	)
+	scanConfigID = *(*scanConfigs.Items)[0].Id
```

Comment:
is it a get or a set function?

Reply:
It is get config from the backend and convert it to families config.

Reply:
I'm a bit confused, the same object type is an input an output for the function.
If the purpose is to set the `scanConfig` on the `cliConf` I would split it into "get" and do the "set" outside.
And IMO, the input for the "get" function doesn't need to be the `cliConf` but only what is needed from it.